src/goalServices/changeGoalStatus.ts

```diff
@@ -0,0 +1,51 @@
+import db from '../models';
+
+interface GoalStatusChangeParams {
+  goalId: number;
+  userId: number;
+  newStatus: string;
+  reason: string;
+  context: string;
+}
+
+export default async function changeGoalStatus({
+  goalId,
+  userId,
+  newStatus,
+  reason,
+  context,
+}: GoalStatusChangeParams) {
+  const user = await db.User.findOne({
+    where: { id: userId },
+    attributes: ['id', 'name'],
+    include: [
+      {
+        model: db.Role,
+        as: 'roles',
+        attributes: ['name'],
+        through: {
+          attributes: [],
+        },
+      },
+    ],
+  });
+
+  const goal = await db.Goal.findByPk(goalId);
```

Comment:
Can these two be done together in a Promise.all that way they can be collected in parallel